independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenArtifactResolver.java

```diff
@@ -127,7 +133,7 @@ private MavenArtifactResolver(Builder builder) throws AppModelResolverException
         if(builder.repoHome != null) {
             final MavenLocalRepositoryManager appCreatorLocalRepoManager = new MavenLocalRepositoryManager(
                     repoSystem.newLocalRepositoryManager(newSession, new LocalRepository(builder.repoHome.toString())),
-                    Paths.get(MavenRepoInitializer.getLocalRepo(MavenRepoInitializer.getSettings())));
+                    builder.localRepoHome==null ? Paths.get(MavenRepoInitializer.getLocalRepo(MavenRepoInitializer.getSettings())) : builder.localRepoHome);
```

Comment:
I think it should just be `builder.repoHome`.

Reply:
but then you are using the same dir as the remote repository and the local cache. Maven treats that differently internally.

if you as a user want it to be the same you should just do that when calling the builder.

Reply:
There is no remote repo here. This is a local repo config. Now that I'm looking into how it's implemented, the benefit of having a delegate was that installed artifacts would end up in `repoHome` instead of the user local one. That was useful for me at the time.
We could keep this as a feature but it shouldn't be the default behavior.

I propose to remove `localRepoHome` and introduce a boolean e.g. `fallbackToUserRepo` which if set to true (the default should be false) would trigger the current behavior, otherwise `repoHome` should be used as the only local repo.

Reply:
i.e. this should be the default local repo manager if `repoHome` isn't null: `repoSystem.newLocalRepositoryManager(newSession, new LocalRepository(builder.repoHome.toString()))`

Reply:
if the user wants delegation then its the current `new MavenLocalRepositoryManager(
                    repoSystem.newLocalRepositoryManager(newSession, new LocalRepository(builder.repoHome.toString())),
                    Paths.get(MavenRepoInitializer.getLocalRepo(MavenRepoInitializer.getSettings())));`

Reply:
as long as you are sure we are not by accident changing maven resolutoin logic for everything else but this one test then i'm fine - if unsure i suggest we keep the change for 1.0 absolute minimal and then in 1+ look at change the logic.

Reply:
I'm sure.